core/wiki/macros/input.tid

```diff
@@ -0,0 +1,67 @@
+title: $:/core/macros/input
+tags: $:/tags/Macro
+
+\define change-input-tab(stateTitle,tag,beforeafter,defaultState)
+<$set name="tabsList" filter="[all[shadows+tiddlers]tag<__tag__>!has[draft.of]]">
+<$vars currentState={{{ [<__stateTitle__>!is[missing]get[text]] ~[<__defaultState__>] }}} firstTab={{{ [enlist<tabsList>nth[1]] }}} lastTab={{{ [enlist<tabsList>last[]] }}}>
+<$set name="nextTab" value={{{ [all[shadows+tiddlers]tag<__tag__>!has[draft.of]$beforeafter$<currentState>] ~[[$beforeafter$]removeprefix[after]suffix[]addprefix<firstTab>] ~[[$beforeafter$]removeprefix[before]suffix[]addprefix<lastTab>] }}}>
+<$action-setfield $tiddler=<<__stateTitle__>> text=<<nextTab>>/>
+</$set>
+</$vars>
+</$set>
+\end
+
+\define accept-input-actions(editTiddler,acceptActions)
+<$vars selectedDropDownItem={{{ [<__editTiddler__>get[text]] }}}>
+$acceptActions$
+<$action-deletetiddler $tiddler=<<__editTiddler__>>/>
+</$vars>
+\end
+
+\define clear-input-field(editTiddler,saveTiddler,refreshTiddler,refreshQualifier)
+<$action-setfield $tiddler=<<__refreshTiddler__>> refresh-qualifier=<<__refreshQualifier__>>/>
+<$action-setfield $tiddler=<<__refreshTiddler__>> text=<<__refreshQualifier__>>/>
+<$action-deletetiddler $tiddler=<<__editTiddler__>>/>
+<$action-deletetiddler $tiddler=<<__saveTiddler__>>/>
+<$action-setfield $tiddler=<<__refreshTiddler__>> text=""/>
+\end
+
+\define clear-input-actions(editTiddler,saveTiddler,refreshTiddler,clearEmptyActions,refreshQualifier)
+<$list filter="[<__editTiddler__>!is[missing]get[text]minlength[1]] [<__saveTiddler__>!is[missing]get[text]minlength[1]] +[limit[1]]" variable="ignore" emptyMessage="""$clearEmptyActions$""">
+<$macrocall $name="clear-input-field" editTiddler=<<__editTiddler__>> saveTiddler=<<__saveTiddler__>> refreshTiddler=<<__refreshTiddler__>> refreshQualifier=<<__refreshQualifier__>>/>
+</$list>
+\end
+
+\define get-dropdown-list()
+$(dropDownFilter1)$ $(dropDownFilter2)$
+\end
+
+\define select-dropdown-actions(beforeafter,reverse,editTiddler,saveTiddler,refreshTiddler,filter1,filter2,filterMinLength)
+<$list filter="[<__saveTiddler__>get[text]minlength[$filterMinLength$]] [[$filterMinLength$]removeprefix[0]suffix[]]">
+<$vars userInput={{{ [<__saveTiddler__>get[text]] }}} selectedDropDownItem={{{ [<__editTiddler__>get[text]] }}}>
+<$set name="dropDownFilter1" filter=<<__filter1__>>>
+<$set name="dropDownFilter2" filter=<<__filter2__>>>
+<$set name="dropDownList" filter=<<get-dropdown-list>>>
+<$list filter="[enlist<dropDownList>] +[$beforeafter$<selectedDropDownItem>] ~[enlist<dropDownList>$reverse$nth[1]]" variable="nextTag">
+<$action-setfield $tiddler=<<__editTiddler__>> text=<<nextTag>>/>
+<$action-deletetiddler $tiddler=<<__refreshTiddler__>>/>
+<$action-setfield $tiddler=<<__refreshTiddler__>> refresh-qualifier={{{ [[$(tv-refresh-input-qualifier)$]] }}}/>
+<$action-setfield $tiddler=<<__refreshTiddler__>> text=<<nextTag>>/>
```

Comment:
theses two lines make the input field refresh after its text has changed through up/down-arrow selections